README.md

```diff
@@ -14,3 +14,22 @@ Documentation:
 
 Test status (most recent release):
 [![PkgEval](https://juliaci.github.io/NanosoldierReports/pkgeval_badges/S/SpecialFunctions.svg)](https://juliaci.github.io/NanosoldierReports/pkgeval_badges/report.html)
+
+## Upgrading from SpecialFunctions 1
+
+SpecialFunctions 2 has only a [single breaking change](https://github.com/JuliaMath/SpecialFunctions.jl/pull/297):
+The removal of the type piracy `Base.factorial(x::Number) = gamma(x + 1)`.
+For most users this change will not break anything but for users of `factorial` it might.
+If you want to upgrade from SpecialFunctions 1 to SpecialFunctions 2 we recommend:
+
+- If your code does not use `factorial` then update the compat entry for SpecialFunctions to e.g. `"1.8.1, 2"`.
+- If your code does use `factorial` then check for all occurrences of `factorial`:
+
+  - If `factorial` is called on an `Integer`, keep `factorial`,
+  - Otherwise replace `factorial(x)` with a call to `gamma(x + 1)`.
+
+  Afterwards update the compat entry for SpecialFunctions and check that your package works with SpecialFunctions 2.
+
+**Be warned**:
```

Comment:
This warning seems like overkill to me; the odds of this breaking an unrelated package that didn't have an explicit SpecialFunctions dependency (and hence would let Pkg know about the required version), seems remote.

For "random things" to break, you would have to have a situation like: Package A uses both SpecialFunctions and package B, and package B calls `factorial` on a non-`Integer` type, so upgrading package A's dependency "breaks" package B.  But in that situation package B would be broken anyway, since without the SpecialFunctions dependency its `factorial` call would throw a `MethodError`.

Reply:
I don't have a strong opinion here, I mainly copied the suggestion in the issue. I agree that it is an unlikely scenario, mainly because it seems not many packages call `factorial` with non integers, but I think your example is a bit misleading here.

As far as I can see, the problematic situation arises if a package A calls `factorial` on non-integers and depends on a package B that loads SpecialFunctions but does not depend on SpecialFunctions itself. This is possible since SpecialFunctions extended `Base.factorial`. However, if package B updates its compat entries to SpecialFunctions 2 and makes this update available in a non-breaking release (e.g., because it does not use `factorial` at all), then `Pkg.update()` will break package A: The type piratical extension of `Base.factorial` is not available anymore but since package A does not depend on SpecialFunctions directly it does not limit SpecialFunctions to < 2.

Reply:
I have toned it down a bit.